tests/unit/test_bids_utils.py

```diff
@@ -81,3 +86,66 @@ def test_bidsmap_scans_run_plus():
     scans = [("1", "run+"), ("2", "run+"), ("3", "run+")]
 
     assert bidsmap_scans(scans) == [("1", "run-01"), ("2", "run-02"), ("3", "run-03")]
+
+
+def test_bidsify_dicom_headers(mocker):
+    """Test bidsify_dicom_headers without ProtocolName field"""
+    dataset = mocker.MagicMock()
+    dataset.__contains__.return_value = False
+
+    dcmread = mocker.patch("pydicom.dcmread")
+    dcmread.return_value = dataset
+
+    bidsify_dicom_headers("filename", "foo")
+
+    assert dataset.data_element.called == False
+
+
+def test_bidsify_dicom_headers_with_protocol_name(mocker):
+    """Test bidsify_dicom_headers with ProtocolName match"""
+    series_description = "foo"
+
+    dataset = mocker.MagicMock()
+    dataset.__contains__.return_value = True
+    dataset.data_element.return_value = mocker.Mock(value=series_description)
+
+    dcmread = mocker.patch("pydicom.dcmread")
+    dcmread.return_value = dataset
+
+    bidsify_dicom_headers("filename", series_description)
+
+    dataset.data_element.assert_called_once_with("ProtocolName")
+
+
+def test_bidsify_dicom_headers_with_protocol_name_mismatch(mocker):
+    """Test bidsify_dicom_headers with ProtocolName mismatch"""
+    series_description = "foo"
+    protocol_name_mock = mocker.Mock(value="bar")
+    series_description_mock = mocker.Mock(value="quux")
+
+    side_effect = [
+        protocol_name_mock,      # 1st call to check if the ProtocolName
+        protocol_name_mock,      # 2nd call to set the ProtocolName
+        series_description_mock, # 3rd call to set SeriesDescription
+    ]
+
+    dataset = mocker.MagicMock()
+    dataset.__contains__.return_value = True
+    dataset.data_element.side_effect = side_effect
```

Comment:
6. In the case where `dataset.data_element("ProtocolName").value` does not already match `series_description` we need to make a couple more calls to `data_element()` in order to change the values to the new `series_description`. We can accomplish that with `side_effect`. 

`side_effect` allows your mocks to throw exceptions, or return different items for different calls. In our case we make three calls to `data_element()`. The first one is in the if statement, the second changes `dataset.data_element("ProtocolName").value` to the new `series_description` and the last changes `dataset.data_element("SeriesDescription").value` to the new `series_description`. In order to test that both the `SeriesDescription` and `ProtocolName` data elements got their values updated, I created two seperate mock objects and have them returned in the side effects